### DISCUSSION TITLE
Show a field or a other depend of value

### DISCUSSION THREAD

**Discussion by Grazulex** [filamentphp/filament#3329] [Help] [answered]
Hello,

I have 2 tables : 
### metas
* id
* name
* is_color
### meta_values
* id
* id_meta
* value

In my 'meta_values' form I have 2 fields ;

```
                Forms\Components\Select::make('meta_id')
                    ->options(CategoryMeta::all()->pluck('name', 'id'))
                    ->searchable()
                    ->required(),
                Forms\Components\ColorPicker::make('value')
                    ->required(),
                Forms\Components\TextInput::make('value')
                    ->required(),
```
The second field 'value' need to be a ColorPicker if the meta choiced on the first field is a meta with a "is_color" attibute on true OR the second field 'value' need to be a TextInput if the meta choiced on the first field is a meta without a 'is_color' attribute (false).

Can I show a field or a other one with this type of condition ?

Thanks

**Comment by danharrin** [comment, marked as answer]
`->visible(fn (Closure $get) => $get('meta_id') === 'is_color')`

https://filamentphp.com/docs/2.x/forms/advanced#using-closure-customisation

But be aware, having fields with duplicate names can sometimes result in unexpected behaviour, I think it's fine here though

**Reply by danharrin** [reply]
Ah also, make sure the select is `reactive()`

**Comment by Grazulex** [comment]
Thanks
